src/states/node.rs

```diff
@@ -1457,10 +1457,10 @@ impl Node {
             }
         }
 
-        let peers_needing_tunnel = self.peer_mgr.peers_needing_tunnel();
-        if !peers_needing_tunnel.is_empty() &&
-           self.peer_mgr.potential_tunnel_nodes().contains(&(*public_id.name(), *peer_id)) {
-            for dst_id in peers_needing_tunnel {
+        for (dst_id, peer_name) in self.peer_mgr.peers_needing_tunnel() {
+            if self.peer_mgr
+                .potential_tunnel_nodes(&peer_name)
```

Comment:
The call to `self.peer_mgr.potential_tunnel_nodes().contains(...)` used to be outside the loop and done only if `peers_needing_tunnel` was non-empty to avoid repeating the relatively costly call.  Not a big deal I guess.